controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -560,18 +536,19 @@ func (r *ROSAControlPlaneReconciler) reconcileClusterAdminPassword(ctx context.C
 	return password, nil
 }
 
-func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSAControlPlaneScope) (bool, error) {
+func validateControlPlaneSpec(ocmClient *ocm.Client, rosaScope *scope.ROSAControlPlaneScope) (bool, error) {
 	// reset previous message.
 	rosaScope.ControlPlane.Status.FailureMessage = nil
 
-	version := rosaScope.ControlPlane.Spec.Version
-	isSupported, err := rosaClient.IsVersionSupported(version)
+	versionList, err := ocm.GetVersionMinorList(ocmClient)
```

Comment:
`GetVersionMinorList` doesn't sound like the correct method to use here.
Also can we keep the current `IsVersionSupported()` function or at least port it to rosa repo? `ValidateVersion()` is doing a lot of redundant checks and requires an extra API call to list versions.

Reply:
well, that was my original concern with importing and using rosa. Rosa handles both classic and HCP and therefore does a lot of extra steps that we don't need. + some of the their logic is related to the cli interactive workflow. I don't think we can just use anything they have as is.
the current validation is 100% simpler and more efficient.

Reply:
It's not our place to tell ROSA that their logic is wrong. If you can provide them with better logic, merge it into their repo, we can use it.

Reply:
I never said its wrong, I am saying we have a different use-case here and this solution is not optimal for us.

Reply:
This is the logic that `openshift/rosa` uses to validate versions when users ask for HCP clusters on ROSA. We have the same requirements that they do. Their `ValidateVersion()` function explicitly handles `isSTS, isHostedCP bool`. Our use-case is a strict subset of their use-case. If we believe that the way they handle this subset is inefficient, any improvement to their logic would always be consumable by us.

Reply:
Copying their logic and changing it downstream means that we cannot consume any correctness changes or improvements they make upstream.

Reply:
I am not arguing introducing this to openshift/rosa and then consuming here is. this what I suggested at the start.
I am saying we shouldn't consume inefficient code before the improvement lands upstream.

Reply:
I will open a PR for this.